runtime/neumann/src/lib.rs

```diff
@@ -1155,6 +1156,30 @@ impl_runtime_apis! {
 		fn generate_task_id(account_id: AccountId, provided_id: Vec<u8>) -> Hash {
 			AutomationTime::generate_task_id(account_id, provided_id)
 		}
+
+		fn calculate_optimal_autostaking(principal: i128, collator: AccountId) -> AutostakingResult {
+			let candidate_info = ParachainStaking::candidate_info(collator);
+			let money_supply = Balances::total_issuance() + Vesting::total_unvested_allocation();
+
+			let collator_stake = candidate_info.unwrap().total_counted as i128;
```

Comment:
This might not be a collator. Currently we return some dirty WASM error.  Should this method return a result to provider a cleaner error ("collator does not exist")?